File: app/src/main/java/com/zacharee1/systemuituner/activites/AboutActivity.java
package com.zacharee1.systemuituner.activites;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.method.LinkMovementMethod;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;

import com.zacharee1.systemuituner.R;
import com.zacharee1.systemuituner.utils.BillingUtil;
import com.zacharee1.systemuituner.utils.RecreateHandler;

public class AboutActivity extends AppCompatActivity
{
    private static boolean DARK = false;
    private BillingUtil mBilling;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);

        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        DARK = sharedPreferences.getBoolean("dark_mode", false);

        setTheme(DARK ? R.style.AppTheme_Dark : R.style.AppTheme);

        RecreateHandler.onCreate(this);

        setContentView(R.layout.activity_about);

        mBilling = new BillingUtil(this);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        TextView appCred = findViewById(R.id.app_credit);
        TextView appLangCred = findViewById(R.id.app_lang_credit);

        appCred.setMovementMethod(LinkMovementMethod.getInstance());
        appLangCred.setMovementMethod(LinkMovementMethod.getInstance());
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item)
    {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy()
    {
        RecreateHandler.onDestroy(this);
        super.onDestroy();
    }

    public void onDonatePayPalClicked(View v) {
        BillingUtil.onDonatePayPalClicked(this);
    }

    public void onDonate1Clicked(View v) {
        mBilling.onDonateClicked("donate_1");
    }

    public void onDonate2Clicked(View v) {
        mBilling.onDonateClicked("donate_2");
    }

    public void onDonate5Clicked(View v) {
        mBilling.onDonateClicked("donate_5");
    }

    public void onDonate10Clicked(View v) {
        mBilling.onDonateClicked("donate_10");
    }
}

File: app/src/main/java/com/zacharee1/systemuituner/activites/info/AboutActivity.kt
package com.zacharee1.systemuituner.activites.info

import android.os.Bundle
import android.text.method.LinkMovementMethod
import android.view.MenuItem
import android.widget.TextView
import com.zacharee1.systemuituner.R
import com.zacharee1.systemuituner.activites.BaseAnimActivity
import com.zacharee1.systemuituner.util.BillingUtil
import kotlinx.android.synthetic.main.layout_donate.*

class AboutActivity : BaseAnimActivity() {
    private val billing by lazy { BillingUtil(this) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_about)
        setTitle(R.string.about)

        val appCred = findViewById<TextView>(R.id.app_credit)
        val langCred = findViewById<TextView>(R.id.lang_credit)

        appCred.movementMethod = LinkMovementMethod.getInstance()
        langCred.movementMethod = LinkMovementMethod.getInstance()

        paypal_button.setOnClickListener { onDonatePayPalClicked() }
        donate_1.setOnClickListener { onDonate1Clicked() }
        donate_2.setOnClickListener { onDonate2Clicked() }
        donate_5.setOnClickListener { onDonate5Clicked() }
        donate_10.setOnClickListener { onDonate10Clicked() }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }

        return super.onOptionsItemSelected(item)
    }

    private fun onDonatePayPalClicked() {
        BillingUtil.onDonatePayPalClicked(this)
    }

    private fun onDonate1Clicked() {
        billing.onDonateClicked("donate_1")
    }

    private fun onDonate2Clicked() {
        billing.onDonateClicked("donate_2")
    }

    private fun onDonate5Clicked() {
        billing.onDonateClicked("donate_5")
    }

    private fun onDonate10Clicked() {
        billing.onDonateClicked("donate_10")
    }
}
